rclpy/rclpy/type_hash.py

```diff
@@ -18,6 +18,11 @@ class TypeHash:
 
     _TYPE_HASH_SIZE = 32
 
+    __slots__ = (
+        '_version',
+        '_value',
+    )
+
```

Comment:
i think that important thing is to add the comment https://github.com/ros2/ros2cli/blob/64d216cb8fafef83d046b79ee6294afb06b7c595/ros2cli/ros2cli/xmlrpc/marshal/rclpy.py#L90-L91 depends on this to avoid API break https://github.com/ros2/rclpy/pull/1243?